### ISSUE TITLE
Siri interactions on phone perform wrong action

### ISSUE THREAD

**Issue by teslarowinc** [Kemmey/Kemmey-TeslaWatch-Public#161]
Just purchased the Apple watch app yesterday specifically for the Siri interface.  I made sure that both my watch and iphone are on the most current release.  I just tested the following commands on my watch and the app is unusable for Siri commands as it doesn't do the requested action and gives wrong outcome responses.

1>  Command:  Hey Siri, lock Tesla doors.   Outcome:  It said it locked the doors but actually, it left the doors open and opened the charge port.  This is a show stopper, sorry.
2>  I manually locked the doors, then gave the command:  Hey Siri, unlock Tesla doors.  Outcome:  It flashed the lights, said it unlocked the doors but did not actually unlock the doors.
3>  Command:  Hey Siri, open Tesla trunk.  Outcome:  It said it could not perform this command.

**Comment by Kemmey** [comment]
Yeah, just had this bug reported today - it is fixed in the beta:

https://testflight.apple.com/join/FdzFOLZv

Siri will only be able to unlock the car - not open frunk or trunk currently. 

These are the four supported commands:

https://twitter.com/WatchForTesla/status/1308677565884887041

**Comment by Kemmey** [comment]
If Siri is your main usecase, you really should go for an app with an iOS counterpart that can donate Shortcuts. Then you can have full control of everything with Siri. That is not possible with an independent watch app. 

Do give Stats app or Remote S a try - both are highly recommended! Feel free to ask Apple for a refund of this app.

**Comment by teslarowinc** [comment]
Thanks for the update, I appreciate it and that you have resolved the bug.  I may circle back in the future and retry your app.

**Comment by hunterxdai** [comment]
Love your app!! Any plans to incorporate open frunk and trunk in the future using Siri? 

I have used Stats for quite some time but it started preventing my car to sleep and the developer was not willing to look into it and said it was because I have TeslaFi...I did my own troubleshooting and confirmed that once I uninstall STATs car would start going to sleep normally. It could be that STATs and TeslaFi use different polling algorithm but the fact STATs doesn't have the ability to stop polling or they aren't willing to troubleshoot and blame other software just really pissed me off. My whole purpose using STATs was for its apple watch capability (i.e. open trunk, frunk, unlock chargeport, etc). 

Switched to your app and it had worked flawlessly so far. I know I can open the watch app and click open frunk/trunk but adding frunk and trunk to Siri would make it even better.

**Comment by Kemmey** [comment]
Hey,

First off, I know the stats app dev - he's a really good guy, and the stats app is a really good app. I'm fairly certain the sleep issue is actually related to running multiple apps or services against the vehicle - there is no real way to synchronize wake-up calls between services, except if all of them were using the same one - which Tesla could solve by modernizing their backend :-D That said - I appreciate the sentiments, but watch app for Tesla will also clash with other services - my app is just extremely dumb in its polling - it doesn't try to be fancy and apply learning algorithms - it'll just poll at set intervals and that is that. Simple - dumb - works :-D and it can be entirely disabled - because I don't need to collect data in the app to present - the app can just refresh when woken up. Different apps have different data needs - and therefore different polling needs and different wake cycles. It's just in the nature of these things. 

So Siri - I've implemented the four native intents Apple provides on the watch. That means that Apple ensures these four are properly interpreted by Siri in all languages etc. I'll look into custom intents at some point, but my initial research showed that those require an iOS counterpart app, which I do not have. That might not be correct, and if so, I'll investigate further. 

TL;DR; - Siri: not at this time. Maybe in future if technically possible. 

Cheers,

Kim

**Comment by electronhunter** [comment]
Thanks for the thorough response, really appreciated. I think what you explained about stats app makes a lot of sense. It is a great app and I have been using it alongside TeslaFi for over half a year but probably just noticed the sleep issue recently...It's been working on and off as I look back at TeslaFi logs now. I guess for my preference, TeslaFi + ur Watch App for Tesla is the perfect solution at this point :-D Can't wait for 1.0.8. Having the complication option is good enough for me! Always wanted to have a dedicated watch face for tesla haha. 

One suggestion, making the complication icon mimic the contour of the vehicle would be even better. 

Thanks,
Hunter

**Comment by Kemmey** [comment]
Thanks :-D 

I've actually commissioned an entire icon set for the app - model 3 outlines for frunk and trunk are specifically part of the commission description :-D 

I'm hoping the timing will allow me to ship this along with 1.0.8, maybe as soon as next week.
